File: 4기-코딩테스트-220608/4기 코딩테스트_220608/4기 - 1st 코딩테스트/CheckPangram/CheckPangram_test.js
import assert from 'assert'
import solution from './CheckPangram.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const s = 'abcdefghijklmnopqrstuvwxyz'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(true, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const s = 'a b c d e f g h i j k l m n o p q r s t u v w x y z a b c'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(true, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const s = 'bcdefghijklmnopqrstuvwxyz'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(false, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const s = 'aaaaaaaaaaaaaaaaaaaaaaaaaa'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(false, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const s = 'nasoiegasjflkesklfjaslkdfjsla'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(false, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const s = 'thequickbrownfoxjumpsoverthelazydog'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual(true, result)
    done();
  });
})

File: 4기-코딩테스트-220608/4기 코딩테스트_220608/4기 - 1st 코딩테스트/CheckPangram/CheckPangram_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.medium.CheckPangram.CheckPangram import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        s = 'abcdefghijklmnopqrstuvwxyz'

        # when
        result = solution(s)

        # then
        self.assertEqual(True, result)

    def test_2(self):
        # given
        s = 'a b c d e f g h i j k l m n o p q r s t u v w x y z a b c'

        # when
        result = solution(s)

        # then
        self.assertEqual(True, result)

    def test_3(self):
        # given
        s = 'bcdefghijklmnopqrstuvwxyz'

        # when
        result = solution(s)

        # then
        self.assertEqual(False, result)

    def test_4(self):
        # given
        s = 'aaaaaaaaaaaaaaaaaaaaaaaaaa'

        # when
        result = solution(s)

        # then
        self.assertEqual(False, result)

    def test_5(self):
        # given
        s = 'nasoiegasjflkesklfjaslkdfjsla'

        # when
        result = solution(s)

        # then
        self.assertEqual(False, result)

    def test_6(self):
        # given
        s = 'thequickbrownfoxjumpsoverthelazydog'

        # when
        result = solution(s)

        # then
        self.assertEqual(True, result)


if __name__ == '__main__':
    unittest.main()
